### PR TITLE
docs: update upload docs

### LINKED ISSUE #29698: [Upload docs issues]The maxCount property of the Upload component doesn't appear in the docs API Property.
- [ ] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[https://ant.design/components/upload/#API](https://ant.design/components/upload/#API)

### Steps to reproduce
https://ant.design/components/upload/#API

### What is expected?
Add this property in the ant design docs

### What is actually happening?
Add this property in the ant design docs

| Environment | Info |
|---|---|
| antd | 4.13.1 |
| React | 17.0 |
| System | macOs |
| Browser | chrome |

### PR DESCRIPTION
**Pull request by ericlee33**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [x] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link
close https://github.com/ant-design/ant-design/issues/29698

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | update upload docs |
| 🇨🇳 Chinese | 更新upload组件文档 |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/upload/index.en-US.md [hunk 1]
```diff
@@ -35,6 +35,7 @@ Uploading is the process of publishing information (web pages, text, pictures, v
 | listType | Built-in stylesheets, support for three types: `text`, `picture` or `picture-card` | string | `text` |  |
 | method | The http method of upload request | string | `post` |  |
 | multiple | Whether to support selected multiple file. `IE10+` supported. You can select multiple files with CTRL holding down while multiple is set to be true | boolean | false |  |
+| maxCount | Limit the number of uploaded files. Will replace current one when `maxCount` is `1` | number | - |  |
```
Comment (xrkffgg) [inline components/upload/index.en-US.md:38]: 按字母顺序加哈

Comment (ericlee33) [reply components/upload/index.en-US.md]: 👌，已更换顺序

File: components/upload/index.en-US.md [hunk 2]
```diff
@@ -33,6 +33,7 @@ Uploading is the process of publishing information (web pages, text, pictures, v
 | isImageUrl | Customize if render &lt;img /> in thumbnail | (file: UploadFile) => boolean | [(inside implementation)](https://github.com/ant-design/ant-design/blob/4ad5830eecfb87471cd8ac588c5d992862b70770/components/upload/utils.tsx#L47-L68) |  |
 | itemRender | Custom item of uploadList | (originNode: ReactElement, file: UploadFile, fileList?: object\[]) => React.ReactNode | - | 4.7.0 |
 | listType | Built-in stylesheets, support for three types: `text`, `picture` or `picture-card` | string | `text` |  |
+| maxCount | Limit the number of uploaded files. Will replace current one when `maxCount` is `1` | number | - |  |
```
Comment (xrkffgg) [inline components/upload/index.en-US.md:36]:
```suggestion
| maxCount | Limit the number of uploaded files. Will replace current one when `maxCount` is `1` | number | - | 4.10.0 |
```

File: components/upload/index.zh-CN.md [hunk 3]
```diff
@@ -34,6 +34,7 @@ cover: https://gw.alipayobjects.com/zos/alicdn/QaeBt_ZMg/Upload.svg
 | isImageUrl | 自定义缩略图是否使用 &lt;img /> 标签进行显示 | (file: UploadFile) => boolean | [(内部实现)](https://github.com/ant-design/ant-design/blob/4ad5830eecfb87471cd8ac588c5d992862b70770/components/upload/utils.tsx#L47-L68) |  |
 | itemRender | 自定义上传列表项 | (originNode: ReactElement, file: UploadFile, fileList?: object\[]) => React.ReactNode | - | 4.7.0 |
 | listType | 上传列表的内建样式，支持三种基本样式 `text`, `picture` 和 `picture-card` | string | `text` |  |
+| maxCount | 限制上传数量。当为 1 时，始终用最新上传的文件代替当前文件 | number | - |  |
```
Comment (xrkffgg) [inline components/upload/index.zh-CN.md:37]:
```suggestion
| maxCount | 限制上传数量。当为 1 时，始终用最新上传的文件代替当前文件 | number | - | 4.10.0 |
```

**Pull request merged**